Add case of 0 elements for functions accepting any number of arguments and give length limit (#25387)
mdn/content: files/en-us/web/javascript/reference/global_objects/array/array/index.md

## Syntax

```js-nolint
new Array()

> new Array(element0OrLength)

Reviewer: Actually, maybe use an overload?

```js
new Array(element0)
new Array(arrayLength)
```

Otherwise, the parameter description below doesn't make sense.

This overload syntax is consistent with, e.g. the `Date()` constructor.

Author: OK. The behavior of Array with 1 argument was also clarified.

### Exceptions

- {{jsxref("RangeError")}}

>   - : Thrown if there's only one argument (`arrayLength`) that is an integer with a value not between 0 and 2<sup>32</sup> - 1 (inclusive).

Reviewer: ```suggestion
  - : Thrown if there's only one argument (`arrayLength`) that is a number, but its value is not an integer or not between 0 and 2<sup>32</sup> - 1 (inclusive).
```

Author: I made the suggested changes. I also removed the validation that arrayOfOne[0] === "2" since it seems redundant if it is being asserted that the logged value is "2" always.

### Array constructor with a single parameter

Arrays can be created using a constructor with a single number parameter. An array is created with
its `length` property set to that number, and the array elements are empty
slots.

```js
const arrayEmpty = new Array(2);

console.log(arrayEmpty.length); // 2

> console.log(arrayEmpty[0]); // undefined
>                             // undefined displayed, but actually an empty slot

Reviewer: ```suggestion
console.log(arrayEmpty[0]); // undefined; actually, it is an empty slot
```
